### PR TITLE
i18n: Advanced Filters strings

### LINKED ISSUE #285: Advanced Filter: UI translation audit
Advanced Filters uses the order of English or LTR to convey what a filter is doing:

screen shot 2018-08-09 at 12 21 31 pm

and 

screen shot 2018-08-09 at 12 21 40 pm

How does this affect RTL languages? Is reversing the order using CSS sufficient in the second image? How can we effectively translate and present the first?

cc @ryelle 

parent issue: https://github.com/woocommerce/wc-admin/issues/256

### PR DESCRIPTION
**Pull request by psealock**
Add translation and context (including link to screenshot) to Advanced Filters such that translators can determine the appropriate translation and the filter is rendered in natural language order.

Relevant discussion in pxLjZ-4T6-p2 and #285 

## Screenshots

English
screen shot 2018-09-25 at 4 29 02 pm

Japanese
screen shot 2018-09-25 at 4 27 28 pm

## Test

1. Go to `/wp-admin/admin.php?page=wc-admin#/analytics/orders?filter=advanced` and poke around the advanced filters to discover any possible regressions.
2. Change the language to something other than English
3. Manually change the translation in the config line 47, https://github.com/woocommerce/wc-admin/blob/f86c16ca7d3af6c2b21c9cf8c3057e52a8fe25f9/client/analytics/report/orders/config.js#L47
to something else (apologies to any Francophiles):
```
{{rule /}} le Status de l'Ordre {{filter /}}
```
4. See the filter rendered accordingly
screen shot 2018-09-25 at 4 21 01 pm
5. Change the values of the filter and see the fieldset's legend update
screen shot 2018-09-25 at 4 38 48 pm
6. Extra Credit: fire up VoiceOver and ensure all legends and labels are available

Fixes #285

### CODE REVIEW

File: client/analytics/report/orders/config.js [hunk 1]
```diff
@@ -27,17 +27,27 @@ export const filters = [
...
+	'Screen shot for context: https://cloudup.com/cSsUY9VeCVJ';
+
 export const advancedFilterConfig = {
 	status: {
 		labels: {
 			add: __( 'Order Status', 'wc-admin' ),
 			remove: __( 'Remove order status filter', 'wc-admin' ),
 			rule: __( 'Select an order status filter match', 'wc-admin' ),
-			title: __( 'Order Status', 'wc-admin' ),
+			title: _x( 'Order Status', titleContext, 'wc-admin' ),
```
Comment (ryelle) [inline client/analytics/report/orders/config.js]: I think we should tweak this, and have it be the formatting control with placeholders for the other content (using `interpolateComponents` in the filter itself), which will let translators flip the content as needed (and allow adding the Japanese particles)

```
_x(
	'Order Status {{rule /}} {{filter /}}',
	'Filter for order status as a sentence, see screenshot for context: https://cloudup.com/cSsUY9VeCVJ',
	'wc-admin'
);
```

File: client/analytics/report/orders/config.js [hunk 2]
```diff
@@ -27,17 +27,27 @@ export const filters = [
...
 
+const titleContext =
+	'Sentence fragment, logical, refers to filtering by Order Status, Product, Customer, or ' +
+	'Coupon Code by various criteria. Screen shot for context: ' +
+	'https://cloudup.com/cSsUY9VeCVJ';
+
+const ruleContext =
+	'Sentence fragment, logical, "Is/Is Not/Includes/Excludes" refers to searching for ' +
+	'orders matching a chosen Order Status, Product, Customer, or Coupon Code. ' +
+	'Screen shot for context: https://cloudup.com/cSsUY9VeCVJ';
```
Comment (ryelle) [inline client/analytics/report/orders/config.js]: Do you think these should be applied to each translation individually, so it'll correspond to the correct object type/string combo?

> Sentence fragment, logical, "Is" refers to searching for orders matching a chosen order status. Screenshot for context: https://cloudup.com/cSsUY9VeCVJ

> Sentence fragment, logical, "Is Not" refers to searching for orders that don't match a chosen order status. Screenshot for context: https://cloudup.com/cSsUY9VeCVJ

> Sentence fragment, logical, "Includes" refers to orders including a given product(s). Screenshot for context: https://cloudup.com/cSsUY9VeCVJ

etc

**Review by ryelle — COMMENTED** [review]
I saw the CSS suggestion in pxLjZ-4T6-p2, but I think that would be too complex. I've added a suggestion here instead using `interpolateComponents`.

File: client/components/filters/advanced/index.js [hunk 3]
```diff
@@ -131,46 +133,47 @@ class AdvancedFilters extends Component {
...
+		const updatedQuery = getQueryFromActiveFilters( activeFilters, query, config.filters );
 		const match = matchValue === 'all' ? undefined : matchValue;
 		return getNewPath( { ...updatedQuery, match }, path, query );
 	}
 
+	isEnglish() {
+		const { siteLocale } = wcSettings;
+		return /en-/.test( siteLocale );
+	}
+
```
Comment (psealock) [inline client/components/filters/advanced/index.js]: Determining if the language is English here so that fields can be lined up vertically to match the designs:

screen shot 2018-08-01 at 12 12 59 pm

This can only happen reliably when the language is known to be English.

Comment (ryelle) [reply client/components/filters/advanced/index.js]: Maybe we can do the CSS a little differently (in another PR) to remove this restriction (I'm thinking maybe a combo of min-width and flex).

File: client/components/filters/advanced/search-filter.js [hunk 4]
```diff
@@ -41,39 +42,65 @@ class SearchFilter extends Component {
...
+				),
+				filter: (
 					<Search
+						className="woocommerce-filters-advanced__input"
 						onChange={ this.onSearchChange }
 						type={ input.type }
 						placeholder={ labels.placeholder }
 						selected={ selected }
-						ariaLabelledby={ `${ key }-${ instanceId }` }
+						aria-label={ labels.filter }
```
Comment (ryelle) [inline client/components/filters/advanced/search-filter.js]: Search doesn't support aria-label at the moment, but that can also be fixed in another PR

Comment (ryelle) [reply client/components/filters/advanced/search-filter.js]: And now it does #484, so you can ignore this comment 😁

File: client/components/filters/advanced/index.js (hunk 3 above)
Comment (psealock) [reply client/components/filters/advanced/index.js]: I can open a new issue to look into this further. What I have, flex + width: 100%, on the input field ensures that the input takes up the maximum space, and that is working well.

The `isEnglish` check is helpful for the defined width of the first column on the left, eg "Order Status", "Product", "Coupon Code", etc. Without knowing the sentence structure, it would look odd to define that width explicitly.

**Comment by psealock** [comment]
Thanks for all your help with this one @ryelle ! I made the changes requested and commented on the `isEnglish` check comment.

**Review by timmyc — APPROVED** [review]
Looking good and testing out well for me, also looks like prior issues brought up in review have been handled. LGTM
